### PR TITLE
Fixing the update for Sabre 2.1

### PR DESCRIPTION
**Pull request by Henni**
Based on the pull request #159 by @PVince81 I fixed an issue which I described on his pull request.

The reponses of the requests
```
http://localhost:8080/index.php/apps/tasks/tasks/init/all
http://localhost:8080/index.php/apps/tasks/tasks/add
```
were empty because the Sabre\VObject creates recursions in vtodos in form of parent and child relations.

### CODE REVIEW

**Comment by raimund-schluessler** [comment]
Thanks for your effort. However, I still get the warning 
```
Error PHP Call to a member function getParts() on a non-object at \ownCloud\app_development\tasks\controller\helper.php#44
```

I will also try to have a look tomorrow. Regards.

File: controller/helper.php [hunk 1]
```diff
@@ -37,10 +37,10 @@ public static function arrayForJSON($id, $vtodo, $user_timezone, $calendarId){
...
 		$task['calendarid'] = $calendarId;
 		$task['type'] = 'task';
-		$task['name'] = $vtodo->SUMMARY;
-		$task['created'] = $vtodo->CREATED;
-		$task['note'] = $vtodo->DESCRIPTION;
-		$task['location'] = $vtodo->LOCATION;
+		$task['name'] = (string) $vtodo->SUMMARY;
+		$task['created'] = (string) $vtodo->CREATED;
+		$task['note'] = (string) $vtodo->DESCRIPTION;
+		$task['location'] = (string) $vtodo->LOCATION;
```
Comment (raimund-schluessler) [inline controller/helper.php]: I don't quite understand why we have to cast everything to ```string``` here. E.g. the calendar app doesn't do it, but if we don't do it, more or less the complete vtodo is repeated.

@PVince81 Can you enlighten me? :)

File: controller/helper.php [hunk 2]
```diff
@@ -37,10 +37,10 @@ public static function arrayForJSON($id, $vtodo, $user_timezone, $calendarId){
...
 		$task['categories'] = $vtodo->CATEGORIES->getParts();
```
Comment (raimund-schluessler) [inline controller/helper.php]: Changing this to ```$task['categories'] = (string) $vtodo->CATEGORIES;``` solves the problem here.

**Comment by Henni** [comment]
I don't see this issue probably because I'm not using categories.
Are categories actually used in the web frontend or do I need a different frontend like thunderbird to work with categories?

**Comment by raimund-schluessler** [comment]
Weird. Actually it doesn't matter if you use categories or not as they are parsed anyway (I plan to integrate categories into the app, so I implemented they already, but the GUI is yet missing).

Also there are problems with the reminder, the percentage of which a task is complete and the comments. Don't you see them too?

**Comment by Henni** [comment]
I don't see any entries in the log about the categories but I see a message regarding the reminder:
´´´
{"reqId":"hEPmZyA\/fWvsugusTaU2","remoteAddr":"127.0.0.1","app":"PHP","message":"Call to undefined method Sabre\\VObject\\Component\\VAlarm::addProperty() at \/...\/owncloud\/local\/apps\/tasks\/controller\/taskscontroller.php#502","level":3,"time":"2015-05-17T14:43:29+00:00"}
´´´

I'll have a look at it.

File: controller/helper.php (hunk 2 above)
Comment (Henni) [inline controller/helper.php]: Should I replace 
```
$task['categories'] = $vtodo->CATEGORIES->getParts();
```
with
```
$categories = $vtodo->CATEGORIES;
if ($categories){
   $task['categories'] = $categories->getParts();
}
```
Does this resolve your issue?

Comment (raimund-schluessler) [inline controller/helper.php]: It resolves the issue with the categories, yes. I will merge your PR and try to fix all issues I find. So either you replace it or I will replace it in a later commit when all issues are fixed.

File: controller/helper.php (hunk 1 above)
Comment (evert) [inline controller/helper.php]: Hi @raimund-schluessler , there's no real preference. `__toString()` maps back to `getValue()` They are identical in behavior, so I would recommend using whatever you think is the most obvious.

Comment (raimund-schluessler) [inline controller/helper.php]: @evert Thanks for the clarification :) I think I'll go for ```(string)``` as I don't have to check whether e.g. ```$vtodo->LOCATION``` returned and object or not.

**Pull request merged**